velox/exec/HashProbe.cpp

```diff
@@ -1436,18 +1500,342 @@ void HashProbe::setRunning() {
   setState(ProbeOperatorState::kRunning);
 }
 
+bool HashProbe::nonReclaimableState() const {
+  return (state_ != ProbeOperatorState::kRunning) || nonReclaimableSection_ ||
+      (inputSpiller_ != nullptr);
+}
+
+void HashProbe::ensureOutputFits() {
+  if (!spillEnabled()) {
```

Comment:
if spilling is not enabled, are we guaranteed to have enough reservation for output??